src/components/MapboxMap.tsx

```diff
@@ -184,3 +186,31 @@ function getDefaultCoordinate<T>(result: Result<T>): Coordinate | undefined {
   }
   return yextDisplayCoordinate;
 }
+
+function renderPinComponentByReactVersion(
```

Comment:
```suggestion
const renderReact = (jsxElement: JSX.Element, container: Element) => {
  try {
    import('react-dom/client').then(({ createRoot }) => {
      const root = createRoot(container);
      root.render(jsxElement);
    });
  } catch {
    ReactDOM.render(jsxElement, container);
  }
}
```
this way the logic is generic and reusable and should hopefully get the tests passing. we could even pull this function out into a separate file. the method signature is also almost the same as react 16/17's ReactDOM.render, so it'll be a trivial change to switch over to using it in the code